src/app/src/pages/eval/components/ConfirmEvalNameDialog.tsx

```diff
@@ -0,0 +1,149 @@
+import { useState, useEffect, useRef } from 'react';
+import WarningIcon from '@mui/icons-material/Warning';
+import Alert from '@mui/material/Alert';
+import Button from '@mui/material/Button';
+import CircularProgress from '@mui/material/CircularProgress';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+
+interface ConfirmEvalNameDialogProps {
+  open: boolean;
+  onClose: () => void;
+  title: string;
+  label: string;
+  currentName: string;
+  actionButtonText: string;
+  helperText?: string;
+  onConfirm: (newName: string) => Promise<void>;
+  // Optional: for large eval warnings (copy only)
+  showSizeWarning?: boolean;
+  itemCount?: number;
+  itemLabel?: string;
+}
+
+export const ConfirmEvalNameDialog = ({
```

Comment:
The props make it seem like this dialog is designed to be generic (e.g. accepting `name` and `actionButtonText`) but it's unclear when and for what it might be used.